### PR TITLE
in user/add function save each successful step into mongo db

### PR DESCRIPTION
**Pull request by StanislavShevchenko**
# Description

in user/add function save each successful step into mongo db
before run each step check om mongo if it is already done not run it

About #119

# How Has This Been Tested?

Register and go to the dashboard

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/db/mongo/user-privat-provider.js [hunk 1]
```diff
@@ -98,6 +98,21 @@ class UserPrivate {
...
 
+  /**
+   * complete Step by identifier and step name
+   *
+   * @param {int} identifier
+   * @param {string} stepName
+   *
+   * @returns {object || null}
+   */
+  async completeStep(identifier, stepName) {
```
Comment (sirpy) [inline src/server/db/mongo/user-privat-provider.js]: this is not the correct way to update a field in mongodb

File: src/server/storage/storage.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    return web3Record
+  }
+  return null
+}
+
+export const updateMarketToken = async (user: any) => {
+  let userDB = await UserDBPrivate.getUser(user.identifier)
+
+  if (!userDB.isCompleted || !userDB.isCompleted.marketToken) {
```
Comment (sirpy) [inline src/server/storage/storage.js]: replace all these double ifs with lodash get(userBD,< object path>)

File: src/server/storage/storage.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const marketToken = await generateMarketToken(user)
+    logger.debug('generate new user market token:', { marketToken, user })
+    if (marketToken) {
+      await UserDBPrivate.updateUser({ identifier: user.identifier, marketToken })
+      await UserDBPrivate.completeStep(user.identifier, 'marketToken')
```
Comment (sirpy) [inline src/server/storage/storage.js]: the completestep is a waste here, whenver you also need to do updateuser, use it to update also the iscompleted

File: src/server/storage/storage.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    return marketToken
+  }
+  return null
+}
+
+export const addUserToTopWallet = async (userRecord: UserRecord) => {
```
Comment (sirpy) [inline src/server/storage/storage.js]: rename to topUserWallet

File: src/server/storage/storageAPI.js [hunk 5]
```diff
@@ -5,35 +5,19 @@ import get from 'lodash/get'
...
-  let encrypted = cipher.update(token, 'utf8', 'base64')
-  encrypted += cipher.final('base64')
-  const ivstring = iv.toString('base64')
-  return `${encrypted}:${ivstring}`
-    .replace(/\+/g, '-')
-    .replace(/\//g, '_')
-    .replace(/=+/g, '')
-}
+import {
+  addUserToWhiteList,
```
Comment (sirpy) [inline src/server/storage/storageAPI.js]: refactor
import addUserSteps from './addUserSteps'

then
addUserSteps.addUserToWhitelist etc...

**Comment by sirpy** [comment]
@AlexeyKosinski this is pending review for a while

File: src/server/db/mongo/user-privat-provider.js (hunk 1 above)
Comment (StanislavShevchenko) [reply src/server/db/mongo/user-privat-provider.js]: made updateOne method

File: src/server/storage/storage.js (hunk 2 above)
Comment (StanislavShevchenko) [reply src/server/storage/storage.js]: done in all place

File: src/server/storage/storage.js (hunk 3 above)
Comment (StanislavShevchenko) [reply src/server/storage/storage.js]: done in all place

File: src/server/storage/storage.js (hunk 4 above)
Comment (StanislavShevchenko) [reply src/server/storage/storage.js]: renamed

File: src/server/storage/storageAPI.js (hunk 5 above)
Comment (StanislavShevchenko) [reply src/server/storage/storageAPI.js]: refactored

**Pull request merged**